pruner/find.go

```diff
@@ -31,7 +31,8 @@ func (s *Service) findPruneableHeaders(
 		return nil, err
 	}
 
-	if lastPruned.Height() == estimatedCutoffHeight {
+	// note that GetRangeByHeight will fetch the range (lastPruned.Height(): estimatedCutoffHeight)
+	if lastPruned.Height() >= estimatedCutoffHeight-1 {
```

Comment:
The condition was making me feel uneasy, and I kept trying to figure out what was off until I had an "aha" moment. According to the `GetRangeByHeight` doc: [lastPruned.Height()+1:to), `to` is not included in the returned range. To prune it, we should request the range [lastPruned.Height():estimatedCutoffHeight+1) from [GetRangeByHeight here](https://github.com/celestiaorg/celestia-node/blob/0da16ddfcf25ea98d41602b59850f462f4df83c3/pruner/find.go#L42). So, to prune `estimatedCutoffHeight`, the condition should allow `lastPruned.Height() < estimatedCutoffHeight`. We can achieve this by removing the `-1` and modifying linked code.
```suggestion
	if lastPruned.Height() >= estimatedCutoffHeight {
```

Reply:
We use the store implementation of `Getter` iface here which includes an additional check 

```
	// as the requested range is non-inclusive in the end[from;to), we need to compare
	// `from` with `to-1`
	if from > to-1 {
		return nil, fmt.Errorf("header/store: invalid range(%d,%d)", from, to-1)
	}

```

where `from` would be lastPruned.Height()+1 and `to` would be `estimatedCutoffHeight`

So if we have a situation where `lastPruned.Height()+1` = `estimatedCutoffHeight`, it would hit this condition and error out as `lastPruned.Height()+1 > estimatedCutoffHeight-1`.

I agree this is very ugly bc it forces the caller to be aware of the detail that `GetRangeByHeight` will error out if a range of 1 header is attempted. The fix should be in header `store` itself though where it should allow a range request of amount `1` to happen as that's still a valid range.

Until that fix is in go-header, however, we should fix this so pruning logs don't keep erroring. I can add a TODO here + make an issue.

Reply:
https://github.com/celestiaorg/go-header/pull/227

Reply:
https://github.com/celestiaorg/celestia-node/pull/3828/commits/e7fc9c5af0fb4a43392028373304099ce9250629

this is the better fix

Reply:
Looks like mentioned commit isn't a part of this PR, am I wrong?